net/src/packet/utils.rs

```diff
@@ -0,0 +1,62 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Open Network Fabric Authors
+//
+//! Packet higher-level methods to allow for code reuse
+
+use std::net::IpAddr;
+
+use crate::eth::Eth;
+use crate::eth::ethtype::EthType;
+use crate::eth::mac::Mac;
+use crate::headers::Net::{Ipv4, Ipv6};
+use crate::headers::{TryEth, TryIp};
+use crate::ip::NextHeader;
+use crate::packet::Packet;
+use crate::packet::PacketBufferMut;
+
+impl<Buf: PacketBufferMut> Packet<Buf> {
+    /// Get the destination mac address of a [`Packet`]
+    /// Returns None if the packet does not have Ethernet header
+    pub fn eth_destination(&self) -> Option<Mac> {
+        self.try_eth().map(|eth| eth.destination().inner())
+    }
+
+    /// Get the source mac address of a [`Packet`]
+    /// Returns None if the packet does not have Ethernet header
+    pub fn eth_source(&self) -> Option<Mac> {
+        self.try_eth().map(|eth| eth.source().inner())
+    }
+
+    /// Get the source mac address of a [`Packet`]
+    /// Returns None if the packet does not have Ethernet header
+    pub fn eth_type(&self) -> Option<EthType> {
```

Comment:
The comment is a copy-paste from the previous method, it needs to be updated for `eth_type()`.